x-pack/legacy/plugins/ml/server/models/data_frame/transforms.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { callWithRequestType } from '../../../common/types/kibana';
+import {
+  DATA_FRAME_TRANSFORM_STATE,
+  DataFrameTransformEndpointData,
+} from '../../../public/data_frame/pages/transform_management/components/transform_list/common';
+
+export function transformServiceProvider(callWithRequest: callWithRequestType) {
+  async function deleteTransform(transformId: string) {
+    return callWithRequest('ml.deleteDataFrameTransform', { transformId });
+  }
+
+  async function stopTransform(options: {
+    transformId: string;
+    force: boolean;
+    waitForCompletion: boolean;
+  }) {
+    return callWithRequest('ml.stopDataFrameTransform', options);
+  }
+
+  async function startTransform(options: { transformId: string; force: boolean }) {
+    return callWithRequest('ml.startDataFrameTransform', options);
+  }
+
+  async function deleteTransforms(transformsInfo: DataFrameTransformEndpointData[]) {
+    const results = [];
+
+    for (const transformInfo of transformsInfo) {
+      try {
+        if (transformInfo.state === DATA_FRAME_TRANSFORM_STATE.FAILED) {
+          await stopTransform({
+            transformId: transformInfo.id,
+            force: true,
+            waitForCompletion: true,
+          });
+        }
+
+        await deleteTransform(transformInfo.id);
+        results.push({ id: transformInfo.id, success: true });
+      } catch (e) {
+        results.push({ id: transformInfo.id, success: false, error: JSON.stringify(e) });
```

Comment:
this might need to handle timeouts in a similar way to `deleteJobs`
If one delete action times out, all subsequent actions are aborted.

Reply:
Yeah, good call. Handle timeouts added for start/stop/delete added here: 5f0e7c3bf1c54e79732bc46364bef460a58400ab